src/auction/src/orders.mo

```diff
@@ -381,20 +391,26 @@ module {
 
       // validate and prepare placements
       for (i in placements.keys()) {
-        let (ordersService, (assetId, volume, price), ordersDelta, oppositeOrdersDelta) = switch (placements[i]) {
+        let (ordersService, (assetId, volume, pr), ordersDelta, oppositeOrdersDelta) = switch (placements[i]) {
           case (#ask(args)) (asks, args, asksDelta, bidsDelta);
           case (#bid(args)) (bids, args, bidsDelta, asksDelta);
         };
+        var price = pr;
 
         // validate asset id
         if (assetId == quoteAssetId or assetId >= assets.nAssets()) return #err(#placement({ index = i; error = #UnknownAsset }));
 
         // validate order volume and price
         let assetInfo = assets.getAsset(assetId);
-        if (ordersService.isOrderLow(assetId, assetInfo, volume, price)) return #err(#placement({ index = i; error = #TooLowOrder }));
-        if (not validatePriceDigits(price)) {
-          return #err(#placement({ index = i; error = #PriceDigitsOverflow({ maxDigits = priceMaxDigits }) }));
+        switch (roundPriceDigits(price)) {
```

Comment:
Can we do 
```
let ?price = roundPriceDigits(pr) else return #err(..)
```
Then we don't need a var assignment first.

Reply:
thanks, updated